wgctrl-rs/src/device.rs

```diff
@@ -103,14 +109,137 @@ pub struct DeviceInfo {
     pub(crate) __cant_construct_me: (),
 }
 
+type RawInterfaceName = [c_char; libc::IFNAMSIZ];
+
+/// The name of a Wireguard interface device.
+#[derive(PartialEq, Eq, Clone, Copy)]
+pub struct InterfaceName(RawInterfaceName);
+
+impl TryFrom<&str> for InterfaceName {
+    type Error = InvalidInterfaceName;
+
+    fn try_from(name: &str) -> Result<Self, InvalidInterfaceName> {
```

Comment:
Might be worth adding a rustdoc here referring to the page describing these name limitations (or even just linking to https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/utils.c#n827), if anyone was wondering where this logic came from.

Reply:
I added it in a doc comment on the `from_str` method as a reference.